File: LeetCode_Solutions/LeetCode_Solutions/AddDigits_258.m
//
//  AddDigits_258.m
//  LeetCode_Solutions
//
//  Created by 来自遥远星系的核心巡洋舰 on 16/10/8.
//  Copyright © 2016年 KnightJoker. All rights reserved.
//



#import "AddDigits_258.h"

@implementation AddDigits_258

- (NSInteger) addDigits:(NSInteger)num {

    while (num > 10) {
        num = num /10 + num % 10;
    }
    return num;
}

@end

File: LeetCode_Swift/LeetCode_Swift/AddDigits_258.swift
//
//  AddDigits_258.swift
//  LeetCode_Swift
//
//  Created by Huni on 02/08/2017.
//  Copyright © 2017 KnighhtJoker. All rights reserved.
//

import Foundation

public class AddDigits_258 {
    func addDigits(_ num: Int) -> Int {
        
        let quotient = num / 10
        let remainder = num % 10
        
        if quotient < 1 {
            return num
        } else {
          return (self.addDigits(quotient + remainder))
        }
        
    }
}
